src/vs/workbench/browser/parts/editor/editorGroupView.ts

```diff
@@ -489,7 +491,9 @@ export class EditorGroupView extends Themable implements IEditorGroupView {
 		// Show active editor (intentionally not using async to keep
 		// `restoreEditors` from executing in same stack)
 		return this.doShowEditor(activeEditor, { active: true, isNew: false /* restored */ }, options).then(() => {
-
+			// Update visibility when restored as the editor start as visible but
+			// could be hidden if panel is maximized
+			this.onDidVisibilityChange(this.layoutService.isVisible(Parts.EDITOR_PART));
```

Comment:
This is needed for the edge case when we have a notebook editor open but with the panel maximized and then reload the window. When the editors are restored they start as visible but are actually hidden as the panel is maximized